libevmasm/GasMeter.cpp

```diff
@@ -128,23 +128,35 @@ GasMeter::GasConsumption GasMeter::estimateMax(AssemblyItem const& _item)
 		case Instruction::CALLCODE:
 		case Instruction::DELEGATECALL:
 		{
-			gas = GasCosts::callGas;
-			if (u256 const* value = classes.knownConstant(m_state->relativeStackElement(0)))
-				gas += (*value);
-			else
+			if (_includeExternalCosts)
+				// We assume that we do not know the target contract and thus, the consumption is infinite.
 				gas = GasConsumption::infinite();
-			if (_item.instruction() == Instruction::CALL)
-				gas += GasCosts::callNewAccountGas; // We very rarely know whether the address exists.
-			int valueSize = _item.instruction() == Instruction::DELEGATECALL ? 0 : 1;
-			if (!classes.knownZero(m_state->relativeStackElement(-1 - valueSize)))
-				gas += GasCosts::callValueTransferGas;
-			gas += memoryGas(-2 - valueSize, -3 - valueSize);
-			gas += memoryGas(-4 - valueSize, -5 - valueSize);
+			else
+			{
+				gas = GasCosts::callGas;
+				if (u256 const* value = classes.knownConstant(m_state->relativeStackElement(0)))
+					gas += (*value);
+				else
+					gas = GasConsumption::infinite();
+				if (_item.instruction() == Instruction::CALL)
+					gas += GasCosts::callNewAccountGas; // We very rarely know whether the address exists.
```

Comment:
Yeah, external cost here means costs incurred by opcodes not part of the current code. Everything that is computed here is charged to the `CALL` opcode itself, i.e. it is subtracted before the gas is forwarded to the called contract.